modules/cactus-web/src/StatusMessage/StatusMessage.tsx

```diff
@@ -1,73 +1,70 @@
+  NotificationInfo,
+  StatusCheck,
+} from '@repay/cactus-icons'
+import { colorStyle, textStyle } from '@repay/cactus-theme'
 import PropTypes from 'prop-types'
 import React from 'react'
-import styled, { css } from 'styled-components'
+import styled from 'styled-components'
+import { margin, MarginProps } from 'styled-system'
 
-import { textStyle } from '../helpers/theme'
+import { Status as BaseStatus } from '../helpers/status'
 
-export type Status = 'success' | 'warning' | 'error'
+type Status = BaseStatus | 'info'
+type ColorKey = 'successLight' | 'warningLight' | 'errorLight' | 'lightContrast'
 
-export const StatusPropType = PropTypes.oneOf<Status>(['success', 'warning', 'error'])
+const StatusPropType = PropTypes.oneOf<Status>(['success', 'warning', 'error', 'info'])
 
-interface StatusMessageProps extends React.HTMLAttributes<HTMLDivElement> {
+interface StatusMessageProps extends React.HTMLAttributes<HTMLDivElement>, MarginProps {
   status: Status
 }
 
-type StatusMap = { [K in Status]: ReturnType<typeof css> }
+type StatusMap = { [K in Status]: ColorKey }
+type IconMap = { [K in Status]: React.ElementType<any> }
 
 const statusMap: StatusMap = {
-  success: css`
-    background-color: ${(p): string => p.theme.colors.status.background.success};
-  `,
-  warning: css`
-    background-color: ${(p): string => p.theme.colors.status.background.warning};
-  `,
-  error: css`
-    background-color: ${(p): string => p.theme.colors.status.background.error};
-  `,
+  success: 'successLight',
+  warning: 'warningLight',
+  error: 'errorLight',
+  info: 'lightContrast',
+}
+
+const iconMap: IconMap = {
+  success: StatusCheck,
+  warning: NotificationAlert,
+  error: NotificationError,
+  info: NotificationInfo,
 }
 
 const Noop = (): null => null
 
-const StatusMessageBase: React.FC<StatusMessageProps> = ({
+const StatusMessage: React.FC<StatusMessageProps> = ({
   status,
-  className,
   children,
   ...rest
 }): React.ReactElement => {
-  let StatusIcon: React.ElementType<any> = Noop
-  switch (status) {
-    case 'error': {
-      StatusIcon = NotificationError
-      break
-    }
-    case 'warning': {
-      StatusIcon = NotificationAlert
-      break
-    }
-    case 'success': {
-      StatusIcon = StatusCheck
-      break
-    }
-  }
+  const StatusIcon: React.ElementType<any> = iconMap[status] || Noop
   return (
-    <div {...rest} role="alert" className={className}>
+    <StyledStatusMessage {...rest} role="alert" $status={status}>
       <StatusIcon aria-hidden="true" />
       <span>{children}</span>
-    </div>
+    </StyledStatusMessage>
   )
 }
 
-const StatusMessage = styled(StatusMessageBase)`
+const StyledStatusMessage = styled.div<{ $status: Status }>`
   padding: 2px 4px;
   position: relative;
   box-sizing: border-box;
   overflow-wrap: break-word;
   display: inline-block;
-  ${(p) => textStyle(p.theme, 'small')};
-  ${(p) => statusMap[p.status]}
+  ${textStyle('small')};
+  ${(p) => colorStyle(statusMap[p.$status])}
```

Comment:
This works, but involves an extra function call. I'd suggest calling it in the second form instead:
```suggestion
  ${(p) => colorStyle(p, statusMap[p.$status])}
```